clients/drcachesim/scheduler/scheduler.h

```diff
@@ -274,6 +280,17 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
          * #dynamorio::drmemtrace::memtrace_stream_t API instead.
          */
         std::vector<range_t> regions_of_interest;
+        /**
+         * Which inputs the details in this structure apply to, expressed as 0-based
+         * ordinals in the 'readers' vector or in the files opened at 'path' (which
+         * are sorted lexicographically by path) in the containing
+         * #dynamorio::drmemtrace::scheduler_tmpl_t::input_workload_t.  Only one of
+         * 'tids' and 'shards' can be non-empty.  If 'tids' is empty and 'shards' is
```

Comment:
> Only one of 'tids' and 'shards' can be non-empty. 

But it is allowed to specify tids in some input_thread_info_t but shards in others?

Is it a requirement for the sets of tids/shards specified in different input_thread_info_t to be mutually exclusive?

Reply:
> Is it a requirement for the sets of tids/shards specified in different input_thread_info_t to be mutually exclusive?

Looks like later modifiers for the same input will be ignored. Document that?

Reply:
Yes you can mix them.  Documented that only the first is honored for both shard and tids if there are multiple.

Reply:
Correction: only the last is honored; later ones override earlier ones.